### ISSUE TITLE
[Bug] [Shell] [UWP] When using Shell.Current.GoToAsync in OnAppearing to redirect to another page at startup, Shell navigation stops working

### ISSUE THREAD

**Issue by drossoft** [xamarin/Xamarin.Forms#14203]
### Description

When using Shell.Current.GoToAsync in OnAppearing to redirect to another page at startup, Shell navigation stops working.

protected async override void OnAppearing()
{
       base.OnAppearing();
       await Shell.Current.GoToAsync($"//{nameof(AboutPage)}");
}

Attached you can find a demo project created using the Shell with tabs template.

### Steps to Reproduce

1. Create a new project using the Shell Tabs template.

2.  Add ShellItem as the first item in AppShell.xaml:

`<ShellItem Route="LoginPage" FlyoutItemIsVisible="False">`
`        <ShellContent ContentTemplate="{DataTemplate local:LoginPage}"/>`
`    </ShellItem>`

3. Set this properties to LoginPage XAML definition:
     Shell.FlyoutBehavior="Disabled"
     Shell.NavBarIsVisible="False"

4. override LoginPage OnAppearing to redirect to AboutPage:

        protected async override void OnAppearing()
        {
            base.OnAppearing();
            await Shell.Current.GoToAsync($"//{nameof(AboutPage)}");
        }

5. Set UWP as Startup project and run the App.

### Expected Behavior
Navigation is redirected to AboutPage cleaning up the navigation stack.

### Actual Behavior
Redirection does not occur, and also, Shell.Current.GoToAsync does not work anymore. As you can see in the sample project, the LoginCommand also calls GoToAsync but nothing happens.

It works **correctly** on Android.
Have not tested it on iOS.

### Basic Information

- Version with issue: 5.0.0.2012
- Platform Target Frameworks:
  - UWP:  Minimum version: 16299
               Target version:  19041

### Reproduction Link

[ShellGotoIssue.zip](https://github.com/xamarin/Xamarin.Forms/files/6360350/ShellGotoIssue.zip)

**Comment by rachelkang** [comment]
Hi, @drossoft - thanks for submitting this issue! I took a look at your sample and noticed a few missing things. The following steps should help you achieve what you're looking to:

1. Register the route for `AboutPage` - be sure to add the following to your `AppShell`:
```cs
Routing.RegisterRoute(nameof(AboutPage), typeof(AboutPage));
```
2. Remove the `//` preceding the page from your `GoToAsync` call:
```cs
await Shell.Current.GoToAsync($"{nameof(AboutPage)}");
```

You can learn more about the nuances of Shell navigation and related patterns here: https://docs.microsoft.com/xamarin/xamarin-forms/app-fundamentals/shell/navigation#register-detail-page-routes

Here's the updated sample with the updated code: 
[ShellGotoIssue (2).zip](https://github.com/xamarin/Xamarin.Forms/files/6393719/ShellGotoIssue.2.zip)

Hope this helps! :)

**Comment by drossoft** [comment]
Hi Rachel, Thanks for your reply!

You are right, the RegisterRoute was missing from the demo solution. But in my project, it is correct, and it still does not work.

Im using the // to clean up the navigation stack. I tried to remove it but I get this exception:

System.Exception: 'Relative routing to shell elements is currently not supported. Try prefixing your uri with ///: ///DestinationPage'

What made me think it's a bug is that the code works perfectly on Android and it does not work on UWP. I guess it has something to do with something else, I'll keep investigating.

Regards,
Diego

**Comment by rachelkang** [comment]
@drossoft that is strange... if you continue having issues, feel free to reopen this issue or a new one with an updated sample

**Comment by drossoft** [comment]
I've double checked my project code and I've realized that the "Relative routing to shell..." exception was caused by the RegisterRoute not being correct. 

Anyway, the problem is that removing the // the login page is not removed from the stack, and I need that. Why does it work on Android and not on UWP? should I use another approach to remove the login page from the navigation stack?

Thanks again for your help!
